src/lib.rs

```diff
@@ -1,5 +1,25 @@
 #![no_std]
 #![deny(missing_debug_implementations)]
+#![deny(missing_docs)]
+
+//! An experimental Multiboot 2 crate for ELF-64/32 kernels.
+//!
+//! The GNU Multiboot specification aims provide a standardised
+//! method of sharing commonly used information about the host machine at
+//! boot time.
+//!
+//! # Examples
+//!
+//! ```ignore
+//! use multiboot2::load;
+//!
+//! // The Multiboot 2 specification dictates that the machine state after the
+//! // bootloader finishes its job will be that the boot information struct pointer
+//! // is stored in the `EBX` register.
+//! let multiboot_info_ptr: u32;
+//! unsafe { asm!("mov $2, %ebx" : "=r"(multiboot_info_ptr)) };
```

Comment:
Isn't this the wrong way round for AT&T syntax? E.g. should be:
```
asm!("mov %ebx, $2" : "=r"(multiboot_info_ptr));
```

Reply:
I avoid using inline assembly out of preference for *out of line* assembly, and even then I use Intel syntax.

That macro was a shot in the dark so I'd give it a 50/50.

If it turns out to be incorrect could an issue be open and labeled `easy` or `help-wanted` as a hook for newer contributors?